src/components/BrandLogo.astro

```diff
@@ -0,0 +1,64 @@
+---
+export interface Props {
+	size?: `${number}rem` | `${number}px`;
+	shape?: 'circle' | 'rounded';
+	brand: 'buddy' | 'cloudflare' | 'deno' | 'github' | 'gitlab'
+	| 'google-cloud' | 'google-firebase' | 'heroku' | 'layer-zero'
+	| 'microsoft-azure' | 'netlify' | 'render' | 'surge' | 'vercel';
+}
+
+const logos: Record<Props['brand'], { file: string; padding: string; }> = {
```

Comment:
Note to self: I think it might be best to extract this and some of the data in the `services` array in `DeployGuidesNav` to a shared data file.